tasks/init.rb

```diff
@@ -0,0 +1,47 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+#
+# Puppet Task to purge nodes
+# This can only be run against the Puppet Master.
+#
+# Parameters:
+#   * agent_certnames - A comma-separated list of agent certificate names.
+#
+require 'puppet'
+require 'open3'
+
+Puppet.initialize_settings
+
+unless Puppet[:server] == Puppet[:certname]
+  puts 'This task can only be run against the Master (of Masters)'
+  exit 1
+end
+
+def exports(resources)
+  command = "/opt/puppetlabs/puppet/bin/puppet node exports"
+
+  unless resources.to_s.empty?
+    command << " --resources \"#{resources}\""
+  end
+
+  stdout, stderr, status = Open3.capture3(command)
```

Comment:
To make this safer, pass the command and each argument to the command in a comma-separated list. e.g:

```ruby
resources_flag = nil

unless resources.to_s.empty?
  resources_flag = '--resources' 
end

stdout, stderr, status = Open3.capture3('/opt/puppetlabs/puppet/bin/puppet', 'node', 'exports', resources_flag, resources)
```

...Or something like that.